### ISSUE TITLE
Yazi leaves random core dump files

### ISSUE THREAD

**Issue by W-Y-R** [sxyazi/yazi#566]
### What system are you running Yazi on?

Linux Wayland

### What terminal are you running Yazi in?

Kitty 0.32.0

### Yazi version

0.2.1

### Did you try the latest main branch to see if the problem has already been fixed?

Tried, but the problem is still present

### Describe the bug

Sometimes I find random core dump files in my folders, generated by yazi. 

### Expected Behavior

Of course, these files should not appear. Not in every folder, at least.

### To Reproduce

They might occur because of unfinished operations in yazi when closing, but I can not say for sure.

### Configuration

```
{ pkgs, ... }:

{
    programs.yazi = {
        enable = true;
        settings = {
            manager = {
            layout         = [ 1 4 3 ];
            sort_by        = "natural";
            sort_sensitive = true;
            sort_reverse   = false;
            sort_dir_first = true;
            linemode       = "none";
            show_hidden    = true;
            show_symlink   = true;
            };

            preview = {
            tab_size        = 2;
            max_width       = 600;
            max_height      = 900;
            cache_dir       = "";
            ueberzug_scale  = 1;
            ueberzug_offset = [ 0 0 0 0 ];
            };

            tasks = {
            micro_workers = 5;
            macro_workers = 10;
            bizarre_retry = 5;
            };

            log = {
            enabled = false;
            };

            opener = {
                edit = [
                    { exec = "nvim \"$@\""; block = true;  for = "unix"; }
                    { exec = "code \"%*\"";    orphan = true; for = "windows"; }
                ];
            };
        };
        keymap = {
            manager.keymap = [
                { exec = "shell 'dragon -x -i -T \"$1\"' --confirm"; on = [ "<C-d>" ]; }
                { exec = "arrow -1"; desc = "Move cursor up"; on = [ "<Up>" ]; }
                { exec = "arrow 1"; desc = "Move cursor down"; on = [ "<Down>" ]; }
                { exec = "paste"; desc = "Paste the files"; on = [ "p" ]; }
                { exec = "escape"; on = [ "<Esc>" ]; }
                { exec = "quit"; on = [ "q" ]; }
                { exec = "close"; on = [ "<C-q>" ]; }
                { exec = "shell \"$SHELL\" --block --confirm"; on = [ "<C-z>" ]; }

                # Navigation
                { exec = "arrow -1"; on = [ "u" ]; }
                { exec = "arrow 1"; on = [ "e" ]; }

                { exec = "arrow -5"; on = [ "U" ]; }
                { exec = "arrow 5"; on = [ "E" ]; }

                { exec = "leave escape --visual --select"; on = [ "<Left>" ]; }
                { exec = "enter escape --visual --select"; on = [ "<Right>" ]; }

                { exec = "back"; on = [ "N" ]; }
                { exec = "forward"; on = [ "I" ]; }

                { exec = "peek -5"; on = [ "<C-u>" ]; }
                { exec = "peek 5"; on = [ "<C-e>" ]; }

                { exec = "arrow -99999999"; on = [ "g" ]; }
                { exec = "arrow 99999999"; on = [ "G" ]; }

                # Selection
                { exec = "select --state=none arrow 1"; on = [ "<Space>" ]; }
                { exec = "visual_mode"; on = [ "v" ]; }
                { exec = "visual_mode --unset"; on = [ "V" ]; }
                { exec = "select_all --state=true"; on = [ "<C-a>" ]; }
                { exec = "select_all --state=none"; on = [ "<C-r>" ]; }

                # Operation
                { exec = "open"; on = [ "<Enter>" ]; }
                { exec = "open --interactive"; on = [ "<C-Enter>" ]; }
                { exec = "yank escape --visual --select"; on = [ "y" ]; }
                { exec = "yank --cut escape --visual --select"; on = [ "x" ]; }
                { exec = "paste"; on = [ "p" ]; }
                { exec = "paste --force"; on = [ "P" ]; }
                { exec = "link"; on = [ "l" ]; }
                { exec = "link --relative"; on = [ "L" ]; }
                { exec = "remove escape --visual --select"; on = [ "d" ]; }
                { exec = "remove --permanently escape --visual --select"; on = [ "D" ]; }
                { exec = "create"; on = [ "a" ]; }
                { exec = "rename"; on = [ "r" ]; }
                { exec = "shell"; on = [ ";" ]; }
                { exec = "shell --block"; on = [ ":" ]; }
                { exec = "hidden toggle"; on = [ "h" ]; }
                { exec = "search fd"; on = [ "s" ]; }
                { exec = "search rg"; on = [ "S" ]; }
                { exec = "search none"; on = [ "<C-s>" ]; }
                { exec = "jump zoxide"; on = [ "z" ]; }
                { exec = "jump fzf"; on = [ "Z" ]; }

                # Linemode
                { exec = "linemode size"; on = [ "m" "s" ]; }
                { exec = " linemode permissions"; on= [ "m" "p" ]; }
                { exec = "linemode mtime"; on = [ "m" ]; }
                { exec = "linemode none"; on = [ "n" ]; }

                # Copy
                { exec = "copy path"; on = [ "c" "c" ]; }
                { exec = "copy dirname"; on = [ "c" "d" ]; }
                { exec = "copy filename"; on = [ "c" "f" ]; }
                { exec = "copy name_without_ext"; on = [ "c" "n" ]; }

                # Find
                { exec = "find --smart"; on = [ "/" ]; }
                { exec = "find --previous --smart"; on = [ "?" ]; }
                { exec = "find_arrow"; on = [ "-" ]; }
                { exec = "find_arrow --previous"; on = [ "=" ]; }

                # Sorting
                { exec = "sort alphabetical --dir_first"; on = [ "," "a" ]; }
                { exec = "sort alphabetical --reverse --dir_first"; on = [ "," "A" ]; }
                { exec = "sort natural --dir_first"; on = [ "," "n" ]; }
                { exec = "sort natural --reverse --dir_first"; on = [ "," "N" ]; }
                { exec = "sort created --reverse --dir_first"; on = [ "," "c" ]; }
                { exec = "sort created --dir_first"; on = [ "," "C" ]; }
                { exec = "sort modified --reverse --dir_first"; on = [ "," "m" ]; }
                { exec = "sort modified --dir_first"; on = [ "," "M" ]; }
                { exec = "sort size --reverse --dir_first"; on = [ "," "s" ]; }
                { exec = "sort size --dir_first"; on = [ "," "S" ]; }

                # Tabs
                { exec = "tab_create --current"; on = [ "t" ]; }
                { exec = "tab_switch -1 --relative"; on = [ "<C-n>" ]; }
                { exec = "tab_switch 1 --relative"; on = [ "<C-i>" ]; }
                { exec = "tab_switch 0"; on = [ "1" ]; }
                { exec = "tab_switch 1"; on = [ "2" ]; }
                { exec = "tab_switch 2"; on = [ "3" ]; }
                { exec = "tab_switch 3"; on = [ "4" ]; }
                { exec = "tab_switch 4"; on = [ "5" ]; }
                { exec = "tab_switch 5"; on = [ "6" ]; }
                { exec = "tab_switch 6"; on = [ "7" ]; }
                { exec = "tab_switch 7"; on = [ "8" ]; }
                { exec = "tab_switch 8"; on = [ "9" ]; }
                { exec = "tab_swap -1"; on = [ "<" ]; }
                { exec = "tab_swap 1"; on = [ ">" ]; }

                # Tasks
                { exec = "tasks_show"; on = [ "w" ]; }

                # Goto
                { exec = "cd ~"; on = [ "g" "h" ]; }
                { exec = "cd ~/.config"; on = [ "g" "c" ]; }
                { exec = "cd ~/Downloads"; on = [ "g" "d" ]; }
                { exec = "cd /tmp"; on = [ "g" "t" ]; }
                { exec = "cd --interactive"; on = [ "g" "<Space>" ]; }

                # Help
                { exec = "help"; on = [ "~" ]; }

                ];

                tasks.keymap = [
                    { exec = "close"; on = [ "<C-q>" "<Esc>" "w" ]; }
                    { exec = "arrow -1"; on = [ "u" ]; }
                    { exec = "arrow 1"; on = [ "e" ]; }
                    { exec = "inspect"; on = [ "<Enter>" ]; }
                    { exec = "cancel"; on = [ "x" ]; }
                    { exec = "help"; on = [ "~" ]; }
                ];
                select.keymap = [
                    { exec = "close"; on = [ "<C-q>" "<Esc>" ]; }
                    { exec = "close --submit"; on = [ "<Enter>" ]; }
                    { exec = "arrow -1"; on = [ "u" ]; }
                    { exec = "arrow 1"; on = [ "e" ]; }
                    { exec = "arrow -5"; on = [ "U" ]; }
                    { exec = "arrow 5"; on = [ "E" ]; }
                    { exec = "help"; on = [ "~" ]; }
                ];
                input.keymap = [
                    { exec = "close"; on = [ "<C-q>" ]; }
                    { exec = "close --submit"; on = [ "<Enter>" ]; }
                    { exec = "escape"; on = [ "<Esc>" ]; }
                    { exec = "backspace"; on = [ "<Backspace>" ]; }
                    { exec = "insert"; on = [ "k" ]; }
                    { exec = "insert --append"; on = [ "a" ]; }
                    { exec = "visual"; on = [ "v" ]; }
                    { exec = "move -1"; on = [ "n" ]; }
                    { exec = "move 1"; on = [ "i" ]; }
                    { exec = "move -999"; on = [ "N" ]; }
                    { exec = "move 999"; on = [ "I" ]; }
                    { exec = "move -999 insert"; on = [ "<C-n>" ]; }
                    { exec = "move 999 insert --append"; on = [ "<C-i>" ]; }
                    { exec = "backward"; on = [ "b" ]; }
                    { exec = "forward"; on = [ "w" ]; }
                    { exec = "forward --end-of-word"; on = [ "h" ]; }
                    { exec = "delete --cut"; on = [ "d" ]; }
                    { exec = "delete --cut move 999"; on = [ "D" ]; }
                    { exec = "delete --cut --insert"; on = [ "c" ]; }
                    { exec = "delete --cut --insert move 999"; on = [ "C" ]; }
                    { exec = "delete move 1 --in-operating"; on = [ "x" ]; }
                    { exec = "yank"; on = [ "y" ]; }
                    { exec = "paste"; on = [ "p" ]; }
                    { exec = "paste --before"; on = [ "P" ]; }
                    { exec = "undo"; on = [ "l" ]; }
                    { exec = "redo"; on = [ "L" ]; }
                    { exec = "help"; on = [ "~" ]; }
                ];
                help.keymap = [
                    { exec = "escape"; on = [ "<Esc>" ]; }
                    { exec = "close"; on = [ "q" "<C-q>" ]; }
                    { exec = "arrow -1"; on = [ "u" ]; }
                    { exec = "arrow 1"; on = [ "e" ]; }
                    { exec = "arrow -5"; on = [ "U" ]; }
                    { exec = "arrow 5"; on = [ "E" ]; }
                    { exec = "filter"; on = [ "/" ]; }
                ];
                completion.keymap = [
                    { exec = "close"; on = [ "<C-q>" ]; }
                    { exec = "close --submit"; on = [ "<Tab>" ]; }
                    { exec = "arrow -1"; on = [ "<C-u>" ]; }
                    { exec = "arrow 1"; on = [ "<C-e>" ]; }
                    { exec = "help"; on = [ "~" ]; }
                ];
        };
        theme = {
            manager = {
            border_symbol = " ";
            };
        };
    };
}
```

**Comment by sxyazi** [comment]
Please upload your core dump file.

**Comment by sxyazi** [comment]
Is Yazi behaving normally when generating these files - any crashes? I used `lldb` to investigate, and it's possible that the generation of these files is caused by a crash, but I'm not sure about the reason for the crash yet...

**Comment by sxyazi** [comment]
> Yes, it does

It crashes right?

> I presume that it may be because of unfinished operations left when forcibly closing yazi

How do you force quit? Through the system task manager, or the `kill` command, or by pressing `q` in yazi? If exiting from yazi, calling `std::process::exit(0)` without waiting for the remaining tasks/threads to finish is the current way, but I'm not sure if it might lead to the creation of those files.

> but instead it would show something like "2 left" on the bottom right

Press `w` to see what was happening

**Comment by W-Y-R** [comment]
> It crashes right?

Sorry, I was not clear enough. No, it does not crash whatsoever. It works perfectly, yet these core dumps still appear.

> How do you force quit?

I simply kill the console with `killactive` through a keybind. I also start yazi directly through a keybind.
```
bind = $mainMod, E, exec, kitty yazi
bind = $mainMod, W, killactive,
```

> Press w to see what was happening

```
Execute `nvim core` with `/home/user/Documents/core`                                                                   
Execute `nvim core` with `/home/user/Documents/core`  
```

**Comment by sxyazi** [comment]
Sorry, this is definitely a problem that needs to be addressed, but unfortunately, I'm not sure how to fix it at the moment - I can't reproduce it on my system, and I'm not sure what is causing these files to be generated.

Could you please try using the latest Yazi v0.2.2 and, before executing the `killactive` command, terminate Yazi with `pkill yazi` or quit Yazi by pressing `q` within Yazi, then observe for a while to see if the problem still. Maybe we can further narrow down this problem.

**Comment by W-Y-R** [comment]
Sure, I will contact the maintainer of the yazi package from nixpkgs to kindly update the package. Until then, I'll change the keybind as per your instructions.

I'll come back with further info when I'll have it.

**Comment by sxyazi** [comment]
Okay. Also please comment out the following lines during the build so that more debugging information can be seen in case of another core dump:

https://github.com/sxyazi/yazi/blob/087448fd94df495a6a659c86519b8d74668159b7/Cargo.toml#L5-L9

**Comment by sxyazi** [comment]
Please reopen if the problem still

**Comment by bitterhalt** [comment]
It seems to crash every time when  I kill the terminal window where yazi lives.

https://github.com/sxyazi/yazi/assets/95308907/34bfe1db-d5d8-4e8e-9888-a1355eac4480

**Comment by W-Y-R** [comment]
I ended up figuring why the problem occured. It was because of an option I had enabled at some point, which put the core dumps in the same directory as the crashes which were happening.

**Comment by sxyazi** [comment]
Hi, another user feedback that the issue has been fixed in Yazi v0.2.2, see https://github.com/sxyazi/yazi/issues/618#issuecomment-1924718819
